packages/next/server/base-server.ts

```diff
@@ -1528,6 +1528,21 @@ export default abstract class Server {
       return null
     }
 
+    if (isSSG) {
+      // set x-nextjs-cache header to match the header
+      // we set for the image-optimizer
+      res.setHeader(
+        'x-nextjs-cache',
+        isManualRevalidate
+          ? 'REVALIDATED'
+          : cacheEntry.isMiss
+          ? 'MISS'
+          : cacheEntry.isStale
+          ? 'STALE'
+          : 'HIT'
```

Comment:
How about adding assertions to the tests to check for the correct header?

Reply:
Added in https://github.com/vercel/next.js/pull/34826/commits/b6c251f068c6136ec12a53da9d1b3c87f26dbadf